ls1intum/Artemis
# `Programming exercises`: Include automatically assessed exercises directly in course score

### Checklist
#### General


#### Client

### Motivation and Context


Currently the doughnut chart on course overview includes all exercises that are finished (the due date is passed) and marked as included in score. For programming exercises with automatic feedback, the first requirement does not make much sense since the students get their feedback immediately. Therefore, I changed the implementation in that way, that automatically assessed exercises are included in the course score directly after release.
### Description

I added an optional requirement to the service method so that those exercises are included directly.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Programming exercise with release and due date in (near) future and automatic assessment

1. Log in to Artemis
2. Navigate to Course Administration
3. Check the doughnut chart on course overview. Maybe take a screenshot or note down the current score
4. Wait until the exercise is released, reload the page and check that the score computation changed according to the maxPoints of the exercise
5. Participate at the programming exercise. Submit something and wait for the feedback.
6. Check the doughnut chart again and make sure your score gets updated accordingly.
7. Submit multiple times and make sure that the score is not only updated once.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage





| File name | Branch coverage (%) | Line coverage (%) |
| -- | -- | -- |
| `courses-score-calculation.service.ts` | 73.33 | 80.64 |
